src/main/java/gregtech/common/covers/ender/CoverAbstractEnderLink.java

```diff
@@ -0,0 +1,495 @@
+                             @Nullable EntityPlayer player, @NotNull ItemStack itemStack) {
+        super.onAttachment(coverableView, side, player, itemStack);
+        if (player != null) {
+            this.playerUUID = player.getUniqueID();
+        }
+    }
+
+    public void updateColor(String str) {
+        if (str.length() == 8) {
+            this.color = str.toUpperCase();
+            updateLink();
+        }
+    }
+
+    @Override
+    public boolean usesMui2() {
+        return true;
+    }
+
+    @Override
+    public ModularPanel buildUI(SidedPosGuiData guiData, PanelSyncManager guiSyncManager) {
+        var panel = GTGuis.createPanel(this, 176, 192);
+
+        return panel.child(CoverWithUI.createTitleRow(getPickItem()))
+                .child(createWidgets(panel, guiSyncManager))
+                .bindPlayerInventory();
+    }
+
+    protected Column createWidgets(ModularPanel panel, PanelSyncManager syncManager) {
+        var name = new StringSyncValue(this::getColorStr, this::updateColor);
+
+        var entrySelectorSH = createEntrySelector(panel);
+        syncManager.syncValue("entry_selector", entrySelectorSH);
+
+        return new Column().coverChildrenHeight().top(24)
+                .margin(7, 0).widthRel(1f)
+                .child(new Row().marginBottom(2)
+                        .coverChildrenHeight()
+                        .child(createPrivateButton())
+                        .child(createColorIcon())
+                        .child(new TextFieldWidget()
+                                .height(18)
+                                .value(name)
+                                .setPattern(COLOR_INPUT_PATTERN)
+                                .widthRel(0.5f)
+                                .marginRight(2))
+                        .child(createEntrySlot())
+                        .child(new ButtonWidget<>()
+                                .overlay(GTGuiTextures.MENU_OVERLAY)
+                                .background(GTGuiTextures.MC_BUTTON)
+                                .disableHoverBackground()
+                                .addTooltipLine(IKey.lang("cover.generic.ender.open_selector"))
+                                .onMousePressed(i -> {
+                                    if (entrySelectorSH.isPanelOpen()) {
+                                        entrySelectorSH.closePanel();
+                                    } else {
+                                        entrySelectorSH.openPanel();
+                                    }
+                                    Interactable.playButtonClickSound();
+                                    return true;
+                                })))
+                .child(createIoRow());
+    }
+
+    protected abstract PanelSyncHandler createEntrySelector(ModularPanel panel);
+
+    protected abstract IWidget createEntrySlot();
+
+    protected IWidget createColorIcon() {
+        // todo color selector popup panel
```

Comment:
Status on this?

Reply:
i'll probably implement this in a separate pr